src/printer_tree.c

```diff
@@ -0,0 +1,3107 @@
+        ret.node.type = trp_empty_type();
+        ret.indent.btw_name_opts = 0;
+        ret.indent.btw_opts_type = 0;
+        ret.indent.btw_type_iffeatures = 0;
+    }
+    return ret;
+}
+
+/**
+ * @brief Get the correct alignment for the node.
+ *
+ * This function is recursively called itself.
+ * It's like a backend function for a function trp_try_normal_indent_in_node.
+ *
+ * @param[in] node is \<node\> representation.
+ * @param[in] pck contains speciall callback functions for printing.
+ * @param[in] indent contains wrapper and indent in node numbers.
+ * @param[in] mll is max line length.
+ * @param[in,out] cnt counting number of characters to print.
+ * @param[in,out] out is output handler.
+ * @return pair of node and indentation numbers of that node.
+ */
+static struct trt_pair_indent_node
+trp_try_normal_indent_in_node_(struct trt_node node, struct trt_pck_print pck, struct trt_pck_indent indent, size_t mll, size_t *cnt, struct ly_out *out)
+{
+    trp_print_line(node, pck, indent, out);
+
+    struct trt_pair_indent_node ret = {indent.in_node, node};
+
+    if (*cnt <= mll) {
+        /* success */
+        return ret;
+    } else {
+        ret.indent = trp_indent_in_node_place_break(ret.indent);
+        if (ret.indent.type != TRD_INDENT_IN_NODE_FAILED) {
+            /* erase information in node due to line break */
+            ret = trp_first_half_node(node, ret.indent);
+            /* check if line fits, recursive call */
+            *cnt = 0;
+            ret = trp_try_normal_indent_in_node_(ret.node, pck, (struct trt_pck_indent) {indent.wrapper, ret.indent}, mll, cnt, out);
+            /* make sure that the result will be with the status divided
+             * or eventually with status failed */
+            ret.indent.type = ret.indent.type == TRD_INDENT_IN_NODE_FAILED ?
+                    TRD_INDENT_IN_NODE_FAILED : TRD_INDENT_IN_NODE_DIVIDED;
+        }
+        return ret;
+    }
+}
+
+/**
+ * @brief Get the correct alignment for the node.
+ *
+ * @param[in] node is \<node\> representation.
+ * @param[in] pck contains speciall callback functions for printing.
+ * @param[in] indent contains wrapper and indent in node numbers.
+ * @param[in] mll is max line length.
+ * @param[in,out] out is output handler.
+ * @return .type == TRD_INDENT_IN_NODE_DIVIDED - the node does not fit in the line, some indent variable has negative value as a line break sign.
+ * @return .type == TRD_INDENT_IN_NODE_NORMAL - the node fits into the line, all indent variables values has non-negative number.
+ * @return .type == TRD_INDENT_IN_NODE_FAILED - the node does not fit into the line, all indent variables has negative or zero values, function failed.
+ */
+static struct trt_pair_indent_node
+trp_try_normal_indent_in_node(struct trt_node node, struct trt_pck_print pck, struct trt_pck_indent indent, size_t mll, struct ly_out *out)
+{
+    struct trt_pair_indent_node ret = {indent.in_node, node};
```

Comment:
Would prefer for this to be `struct trt_pair_indent_node ret = {.indent = indent.in_node, .node = node};` (and similarly on other places).